rskj-core/src/main/java/co/rsk/signing/ECDSASignerFromDiskKey.java

```diff
@@ -0,0 +1,62 @@
+package co.rsk.signing;
+
+import org.ethereum.crypto.ECKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.util.List;
+
+/**
+ * Signer that reads key from disk.
+ *
+ * Key is identified by a key id.
+ *
+ * Has certain requirements on filesystem permissions
+ * of the file.
+ *
+ * No special authorization is required to sign a message.
+ *
+ * @author Ariel Mendelzon
+ */
+public class ECDSASignerFromDiskKey implements ECDSASigner {
+    private static final Logger logger = LoggerFactory.getLogger("ECDSASignerFromDiskKey");
+
+    private KeyId keyId;
+    private String keyPath;
+
+    public ECDSASignerFromDiskKey(KeyId keyId, String keyPath) {
+        this.keyId = keyId;
+        this.keyPath = keyPath;
+    }
+
+    @Override
+    public boolean canSignWith(KeyId keyId) {
```

Comment:
I think it would be better if the `Signer` object held the key, and you removed the `keyId` parameter from `sign`. This way the signer can only sign with his own key.

Reply:
Please refer to the corresponding documentation to understand the purpose of the key id.